### PR TITLE
Added chunking support

### LINKED ISSUE #142: Support chunking of large log messages
For rather big messages it could make sense to split up a message into multiple smaller chunks that can be logged individually. It would break single messages apart which decreases readability, but at least the whole content could be logged then.

### PR DESCRIPTION
**Pull request by whiskeysierra**
Fixes #142

### CODE REVIEW

File: logbook-core/src/main/java/org/zalando/logbook/ChunkingHttpLogWriter.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+public final class ChunkingHttpLogWriter implements HttpLogWriter {
+
+    private final Pattern pattern;
+    private final HttpLogWriter writer;
+
+    public ChunkingHttpLogWriter(final int size, final HttpLogWriter writer) {
+        this.pattern = Pattern.compile("(?<=\\G.{" + size + "})");
```
Comment (jhorstmann) [inline logbook-core/src/main/java/org/zalando/logbook/ChunkingHttpLogWriter.java]: This looks like a very inefficient way to split a string into fixed length subparts. A better alternative might be to use a `StringReader`.

File: logbook-core/src/main/java/org/zalando/logbook/ChunkingHttpLogWriter.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @Override
+    public void writeResponse(final Correlation<String, String> correlation) throws IOException {
+        for (final String part : split(correlation.getResponse())) {
+            writer.writeResponse(new SimpleCorrelation<>(correlation.getId(), correlation.getRequest(), part));
+        }
+    }
+
+    private Iterable<String> split(final String s) {
```
Comment (jhorstmann) [inline logbook-core/src/main/java/org/zalando/logbook/ChunkingHttpLogWriter.java]: Still creating the whole list up front, I was thinking about something like

    private void splitted(final String string, Consumer<String> consumer) throws IOException {
        final char[] buf = new char[size];
        int len;
        try (final StringReader r = new StringReader(string)) {
            while (-1 != (len = r.read(buf))) {
                consumer.accept(new String(buf, 0, len));
            }
        }
    }

Although you would need a `Consumer` interface that is allowed to throw `IOException` to make it this generic. Or just inline the code twice.

File: logbook-core/src/test/java/org/zalando/logbook/ChunkingHttpLogWriterTest.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        final List<String> precorrelation = captureResponse("Hello");
+        assertThat(precorrelation, contains("Hello"));
+
+    }
+
+    @Test
+    public void shouldWriteResponseInChunksIfLengthExceeded() throws IOException {
+        final List<String> precorrelation = captureResponse("Lorem ipsum dolor sit amet, consectetur adipiscing elit");
+        assertThat(precorrelation,
+                contains("Lorem ipsu", "m dolor si", "t amet, co", "nsectetur ", "adipiscing", " elit"));
```
Comment (AlexanderYastrebov) [inline logbook-core/src/test/java/org/zalando/logbook/ChunkingHttpLogWriterTest.java]: Given response body is usually a plain text 
And log is intended to be read by human
When logbook splits long line into chunks
Then split boundary should be (possibly) located at white-space, within say `size ± delta` interval

Comment (whiskeysierra) [inline logbook-core/src/test/java/org/zalando/logbook/ChunkingHttpLogWriterTest.java]: Good idea. Any idea how to do that efficiently?

Comment (AlexanderYastrebov) [inline logbook-core/src/test/java/org/zalando/logbook/ChunkingHttpLogWriterTest.java]: > Any idea how to do that efficiently?
Leap by size and look around within `± delta` for whitespace

Comment (AlexanderYastrebov) [inline logbook-core/src/test/java/org/zalando/logbook/ChunkingHttpLogWriterTest.java]: After thinking for a while, I have some more points:
1. Split boundary should be within `[size-delta, size]` so chunk is not longer then `size`
2. `delta` should be fixed size of e.g. to average "token" length. I think 10 to 16 characters should be fine
3. Split should happen on whitespaces as well as after common json punctuation characters `,:`

Having said that split algorithm could leap for `size` number of characters and look back `delta` characters for whitespaces and punctuation.

Comment (AlexanderYastrebov) [inline logbook-core/src/test/java/org/zalando/logbook/ChunkingHttpLogWriterTest.java]: I will raise a separate ticket

**Comment by AlexanderYastrebov** [comment]
:+1:

**Pull request merged**